this is what i see in master tablet logs
```
2019-11-01T10:33:11.741769Z 2 [Note] Aborted connection 2 to db: 'unconnected' user: 'root' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:12.126447Z 6 [Note] Aborted connection 6 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:12.128850Z 8 [Note] Aborted connection 8 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:12.204068Z 10 [Note] Aborted connection 10 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:18.477324Z 19 [Note] Semi-sync replication initialized for transactions.
2019-11-01T10:33:18.477924Z 19 [Note] Semi-sync replication enabled on the master.
2019-11-01T10:33:18.478009Z 0 [Note] Starting ack receiver thread
2019-11-01T10:33:18.488530Z 22 [Note] Unknown database 'vt_ks'
2019-11-01T10:33:18.489532Z 23 [Note] Unknown database 'vt_ks'
2019-11-01T10:33:18.490006Z 24 [Note] Unknown database 'vt_ks'
2019-11-01T10:33:18.576666Z 27 [Note] Start binlog_dump to master_thread_id(27) slave_server(1946235800), pos(, 4)
2019-11-01T10:33:18.576700Z 27 [Note] Start semi-sync binlog_dump to slave (server_id: 1946235800), pos(, 4)
2019-11-01T10:33:18.688236Z 28 [Note] Aborted connection 28 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:18.821596Z 35 [Note] Aborted connection 35 to db: 'vt_ks' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:22.227550Z 36 [Note] Aborted connection 36 to db: 'vt_ks' user: 'vt_app' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:22.237013Z 40 [Note] Aborted connection 40 to db: 'vt_ks' user: 'vt_filtered' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:24.112176Z 41 [Note] Aborted connection 41 to db: 'vt_ks' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:33:24.118274Z 0 [ERROR] /usr/local/mysql/bin/mysqld: Got an error reading communication packets
2019-11-01T10:33:54.133641Z 27 [Note] Stop semi-sync binlog_dump to slave (server_id: 1946235800)
2019-11-01T10:33:54.137839Z 27 [Note] Aborted connection 27 to db: 'unconnected' user: 'vt_repl' host: '::1' (failed on flush_net())
2019-11-01T10:34:24.048302Z 17 [Note] Aborted connection 17 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:34:24.050294Z 18 [Note] Aborted connection 18 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:34:24.050735Z 19 [Note] Aborted connection 19 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:34:24.051095Z 20 [Note] Aborted connection 20 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)
2019-11-01T10:34:24.051447Z 21 [Note] Aborted connection 21 to db: 'unconnected' user: 'vt_dba' host: 'localhost' (Got an error reading communication packets)

```

It looks like when I `StopSlave` then master also got stopped.

--- a/go/test/endtoend/cluster/cluster_process.go
+++ b/go/test/endtoend/cluster/cluster_process.go
@@ -57,6 +57,9 @@ type LocalProcessCluster struct {
 
 	//Extra arguments for vtGate
 	VtGateExtraArgs []string
+
+	// To enable SemiSync for vttablets
+	EnableSemiSync bool
 }
 
 // Keyspace : Cluster accepts keyspace to launch it
@@ -186,7 +189,8 @@ func (cluster *LocalProcessCluster) StartKeyspace(keyspace Keyspace, shardNames 
 				cluster.topoProcess.Port,
 				cluster.Hostname,
 				cluster.TmpDirectory,
-				cluster.VtTabletExtraArgs)
+				cluster.VtTabletExtraArgs,
+				cluster.EnableSemiSync)
 			log.Info(fmt.Sprintf("Starting vttablet for tablet uid %d, grpc port %d", tablet.TabletUID, tablet.GrpcPort))
 
 			if err = tablet.vttabletProcess.Setup(); err != nil {
@@ -238,7 +242,7 @@ func (cluster *LocalProcessCluster) StartVtgate() (err error) {
 		cluster.VtgateMySQLPort,
 		cluster.Cell,
 		cluster.Cell,
-    cluster.Hostname,
+		cluster.Hostname,
 		"MASTER,REPLICA",
 		cluster.topoProcess.Port,
 		cluster.TmpDirectory,

--- a/go/test/endtoend/cluster/vttablet_process.go
+++ b/go/test/endtoend/cluster/vttablet_process.go
@@ -56,6 +56,7 @@ type VttabletProcess struct {
 	VtctldAddress               string
 	Directory                   string
 	VerifyURL                   string
+	EnableSemiSync              bool
 	//Extra Args to be set before starting the vttablet process
 	ExtraArgs []string
 
@@ -82,7 +83,6 @@ func (vttablet *VttabletProcess) Setup() (err error) {
 		"-init_keyspace", vttablet.Keyspace,
 		"-init_tablet_type", vttablet.TabletType,
 		"-health_check_interval", fmt.Sprintf("%ds", vttablet.HealthCheckInterval),
-		"-enable_semi_sync",
 		"-enable_replication_reporter",
 		"-backup_storage_implementation", vttablet.BackupStorageImplementation,
 		"-file_backup_storage_root", vttablet.FileBackupStorageRoot,
@@ -90,6 +90,9 @@ func (vttablet *VttabletProcess) Setup() (err error) {
 		"-service_map", vttablet.ServiceMap,
 		"-vtctld_addr", vttablet.VtctldAddress,
 	)
+	if vttablet.EnableSemiSync {
+		vttablet.proc.Args = append(vttablet.proc.Args, "-enable_semi_sync")
+	}
 	vttablet.proc.Args = append(vttablet.proc.Args, vttablet.ExtraArgs...)
 
 	vttablet.proc.Stderr = os.Stderr
@@ -171,7 +174,7 @@ func (vttablet *VttabletProcess) TearDown() error {
 // VttabletProcessInstance returns a VttabletProcess handle for vttablet process
 // configured with the given Config.
 // The process must be manually started by calling setup()
-func VttabletProcessInstance(port int, grpcPort int, tabletUID int, cell string, shard string, keyspace string, vtctldPort int, tabletType string, topoPort int, hostname string, tmpDirectory string, extraArgs []string) *VttabletProcess {
+func VttabletProcessInstance(port int, grpcPort int, tabletUID int, cell string, shard string, keyspace string, vtctldPort int, tabletType string, topoPort int, hostname string, tmpDirectory string, extraArgs []string, enableSemiSync bool) *VttabletProcess {
 	vtctl := VtctlProcessInstance(topoPort, hostname)
 	vttablet := &VttabletProcess{
 		Name:                        "vttablet",
@@ -194,6 +197,7 @@ func VttabletProcessInstance(port int, grpcPort int, tabletUID int, cell string,
 		PidFile:                     path.Join(os.Getenv("VTDATAROOT"), fmt.Sprintf("/vt_%010d/vttable.pid", tabletUID)),
 		VtctldAddress:               fmt.Sprintf("http://%s:%d", hostname, vtctldPort),
 		ExtraArgs:                   extraArgs,
+		EnableSemiSync:              enableSemiSync,
 	}
 
 	if tabletType == "rdonly" {

--- a/go/test/endtoend/tabletmanager/lock_unlock_test.go
+++ b/go/test/endtoend/tabletmanager/lock_unlock_test.go
@@ -97,10 +97,7 @@ func TestStartSlaveUntilAfter(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	fmt.Println("position test") //TODO: Remove this line after testing
-
-	qr := exec(t, masterConn, "insert into t1(id, value) values(1,'a')")
-	fmt.Println(fmt.Sprintf("%v", qr.Rows)) //TODO: Remove this line after testing
+	exec(t, masterConn, "insert into t1(id, value) values(1,'a')")
 	pos1, err := tmcMasterPosition(ctx, masterTabletGrpcPort)
 	if err != nil {
 		t.Fatal(err)
@@ -142,6 +139,11 @@ func TestStartSlaveUntilAfter(t *testing.T) {
 	}
 	checkDataOnReplica(t, replicaConn, `[[VARCHAR("a")] [VARCHAR("b")] [VARCHAR("c")]]`)
 
+	// Strat replication to the replica
+	err = tmcStartSlave(ctx, replicaTabletGrpcPort)
+	if err != nil {
+		t.Fatal(err)
+	}
 	// Clean the table for further testing
 	exec(t, masterConn, "delete from t1")
 }

--- a/go/test/endtoend/tabletmanager/main_test.go
+++ b/go/test/endtoend/tabletmanager/main_test.go
@@ -102,7 +102,7 @@ func TestMain(m *testing.M) {
 			SchemaSQL: sqlSchema,
 			VSchema:   vSchema,
 		}
-		if err = clusterInstance.StartKeyspace(*keyspace, []string{"-80", "80-"}, 1, false); err != nil {
+		if err = clusterInstance.StartUnshardedKeyspace(*keyspace, 1, false); err != nil {
 			return 1
 		}
 

--- a/go/test/endtoend/tabletmanager/main_test.go
+++ b/go/test/endtoend/tabletmanager/main_test.go
@@ -95,6 +95,8 @@ func TestMain(m *testing.M) {
 			"-watch_replication_stream",
 			"-enable_replication_reporter",
 		}
+		// We do not need semiSync for this test case.
+		clusterInstance.EnableSemiSync = false
 
 		// Start keyspace
 		keyspace := &cluster.Keyspace{
@@ -168,6 +170,11 @@ func tmcUnlockTables(ctx context.Context, tabletGrpcPort int) error {
 func tmcStopSlave(ctx context.Context, tabletGrpcPort int) error {
 	vtablet := getTablet(tabletGrpcPort)
 	return tmClient.StopSlave(ctx, vtablet)
+}
+
+func tmcStartSlave(ctx context.Context, tabletGrpcPort int) error {
+	vtablet := getTablet(tabletGrpcPort)
+	return tmClient.StartSlave(ctx, vtablet)
 }
 
 func tmcMasterPosition(ctx context.Context, tabletGrpcPort int) (string, error) {

